azurerm/internal/services/subscription/subscription_resource.go

```diff
@@ -0,0 +1,499 @@
+			tenantId = *resp.TenantID
+		}
+
+		if resp.DisplayName != nil {
+			subscriptionName = *resp.DisplayName
+		}
+
+		state := ""
+		if resp.State == subscriptions.Disabled || resp.State == subscriptions.Warned {
+			state = "Cancelled"
+		}
+
+		if resp.State == subscriptions.Enabled {
+			state = "Active"
+		}
+
+		d.Set("state", state)
+	}
+
+	// (@jackofallops) A subscription's billing scope is not exposed in any way in the API/SDK so we cannot read it back here
+
+	d.Set("subscription_id", subscriptionId)
+	d.Set("subscription_name", subscriptionName)
+	d.Set("tenant_id", tenantId)
+
+	return nil
+}
+
+// (@jackofallops) - Delete here is a misnomer.  The nature of subscriptions is such that they are never truly deleted
+// Deleted here means Cancelled. Cancelled subscriptions are held in this state for  90 days before being purged from
+// active systems.  However, the backend billing systems _never_ remove this data, so once a Subscription ID has been
+// used and purged from active use it can never be recovered nor the UUID reused.
+// Note Cancelling a Subscription leaves it in one of several states, `Disabled` for a Subscription with no Resources or
+// Alias assignments, `Warned` for Cancelled with "something" associated with it.
+func resourceSubscriptionDelete(d *schema.ResourceData, meta interface{}) error {
+	aliasClient := meta.(*clients.Client).Subscription.AliasClient
+	subscriptionClient := meta.(*clients.Client).Subscription.SubscriptionClient
+	client := meta.(*clients.Client).Subscription.Client
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.SubscriptionAliasID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	// Get subscription details for later
+	alias, err := aliasClient.Get(ctx, id.Name)
+	if err != nil || alias.Properties == nil {
+		return fmt.Errorf("could not read Alias %q for Subscription: %+v", id.Name, err)
+	}
+	subscriptionId := ""
+	if subscriptionIdRaw := alias.Properties.SubscriptionID; subscriptionIdRaw != nil {
+		subscriptionId = *subscriptionIdRaw
+	}
+
+	sub, err := client.Get(ctx, subscriptionId)
+	if err != nil {
+		return fmt.Errorf("could not read Subscription details for %q: %+v", subscriptionId, err)
+	}
+
+	subscriptionName := ""
+	if subscriptionNameRaw := sub.DisplayName; subscriptionNameRaw != nil {
+		subscriptionName = *sub.DisplayName
+	}
+
+	if subscriptionName == "" || subscriptionId == "" {
+		return fmt.Errorf("one or both of Subscription Name (%q) and Subscription ID (%q) could not be determined", subscriptionName, subscriptionId)
+	}
+	// remove the alias
+	resp, err := aliasClient.Delete(ctx, id.Name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(resp) {
+			return fmt.Errorf("could not delete Alias %q for Subscription %q (ID: %q): %+v", id.Name, subscriptionName, subscriptionId, err)
+		}
+	}
+
+	// Cancel the Subscription
+	if _, err := subscriptionClient.Cancel(ctx, subscriptionId); err != nil {
```

Comment:
we should likely do this conditionally - checking if this is the last alias for the subscription and locking on the subscription id during create and delete (when re-enabling and cancelling)